lld/test/COFF/arm64x-undef.s

```diff
@@ -0,0 +1,26 @@
+// REQUIRES: aarch64, x86
+// RUN: split-file %s %t.dir && cd %t.dir
+
+// RUN: llvm-mc -filetype=obj -triple=arm64ec-windows sym.s -o sym-arm64ec.obj
+// RUN: llvm-mc -filetype=obj -triple=x86_64-windows sym.s -o sym-x86_64.obj
+// RUN: llvm-mc -filetype=obj -triple=aarch64-windows symref.s -o symref-aarch64.obj
+
+// Check that native object files can't reference EC symbols.
+
+// RUN: not lld-link -machine:arm64x -dll -noentry -out:out.dll symref-aarch64.obj sym-arm64ec.obj \
+// RUN:              2>&1 | FileCheck --check-prefix=UNDEF %s
+// UNDEF:      lld-link: error: undefined symbol: sym
```

Comment:
I think this test would be clearer if we'd also have a positive test, to see that symbols within each namespace can get resolved correctly. I guess that's kinda implicitly tested in all the other existing tests, but it would be nice for clarity here as well.

Reply:
Good idea, I'll add such a test.

Reply:
Done. I also added a test reading both ECSYMBOLS and native symbol indexes from an archive.